### ISSUE TITLE
Cross Entropy AutoEncoder 

### ISSUE THREAD

**Issue by Mirmidion** [zoomin-lee/SemCity#6]
Hi! When going through the repo and trying some things out, one thing stood out to me. I was wondering why the cross-entropy loss was calculated with the logits output of the decoder, instead of the softmax output of the AutoEncoder:

https://github.com/zoomin-lee/SemCity/blob/b405e58a2d5d1e0202f9347eb54d6b7ce8979fdd/encoding/train_ae.py#L63-L69

As to my understanding this currently calculates a cross-entropy that does not fully represent what it is meant to be, and limits the model in converging to the data. Is my reasoning correct, and if so, is there a reason why it is done like this?

ps.
Great and interesting paper! Trying to currently adapt it to my own data and improve it.
